lib/devise/passkeys/controllers/reauthentication_controller_concern.rb

```diff
@@ -100,7 +100,10 @@ def reauthenticate
 
           store_reauthentication_token_in_session
 
-          render json: { reauthentication_token: stored_reauthentication_token }
+          render json: { 
+            reauthentication_token: stored_reauthentication_token,
+            new_csrf_token: form_authenticity_token
```

Comment:
@tcannonfodder : I am not sure yet this is the best approach nor the name is a good one, but this solved the invalid authenticity issue for me.
Let me know if this helps your use case as well.